### ISSUE TITLE
Instance module.ABC.module.XYZ.random_id.suffix data could not be decoded from the state: unsupported attribute "b64". when changing from version 2.3.0 to 3.0.0

### ISSUE THREAD

**Issue by andresvia** [hashicorp/terraform-provider-random#122]
Hi, I have not yet get around lock providers in specific versions using the 0.13 syntax.

However this happened when I had a terraform configuration previously applied with 2.3.0 and then broke when the plugin got the update to 3.0.0.

It is hard for me to add a reproduction right now, but I'll try to add it later.

### Terraform Version
Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed.

```
terraform version

Your version of Terraform is out of date! The latest version
is 0.13.4. You can update by downloading from https://www.terraform.io/downloads.html
Terraform v0.13.0
```

### Affected Resource(s)
Please list the resources as a list, for example:
- random_id

### Terraform Configuration Files

```hcl
# to be added
```

### Debug Output

To be added.

### Panic Output

To be added.

### Expected Behavior

Plan should work.

### Actual Behavior

Plan didn't worked.

### Steps to Reproduce

1. `terraform plan`

```
Linux-x86_64-terraform.bin plan -input=false -out terraform.plan -detailed-exitcode
Error: Invalid resource instance data in state
  on xyz/xyz.tf line 14:
  14: resource "random_id" "abc" {
Instance module.abc.module.xyz1.random_id.abc data could not be
decoded from the state: unsupported attribute "b64".
Error: Invalid resource instance data in state
  on xyz/xyz.tf line 14:
  14: resource "random_id" "abc" {
Instance module.abc.module.xyz2.random_id.abc data could not be
decoded from the state: unsupported attribute "b64".
```

### Important Factoids

Provider not locked in 0.13 style.

**Comment by DaveButland** [comment]
I just experienced the same problem, also running terraform 0.13.0

### Expected Behavior

Terraform ignores or removes the deprecated value in the state file?

### Actual Behavior

Terrform plan fails (see OP)

### Steps to Reproduce

1. terraform apply
```
terraform {
  required_providers {
    random = {
      source  = "hashicorp/random"
      version = "2.3.0"
    }
  }
}

resource "random_id" "secret" {
  byte_length = 8
}
```
2. terraform plan
```
terraform {
  required_providers {
    random = {
      source  = "hashicorp/random"
      version = "3.0.0"
    }
  }
}

resource "random_id" "secret" {
  byte_length = 8
}
```

**Comment by andresvia** [comment]
@DaveButland , looks like this was fixed in 0.13.1 I had to upgrade to 0.13.4 and the problem was no longer there.

See: https://github.com/hashicorp/terraform/issues/25752

I'm closing this.
